ledger/src/blockstore_processor.rs

```diff
@@ -971,7 +1031,18 @@ fn confirm_slot_entries(
     let slot = bank.slot();
     let (entries, num_shreds, slot_full) = slot_entries_load_result;
     let num_entries = entries.len();
-    let num_txs = entries.iter().map(|e| e.transactions.len()).sum::<usize>();
+    let mut entry_indexes = Vec::with_capacity(num_entries);
+    let mut entry_starting_index = progress.num_txs;
```

Comment:
A unit test showing that a few consecutive calls to `confirm_slot_entries` 

1) Properly update the `progress.num_txs` 
2) Return the correct transaction indexes to the consumer

would be good future-proofing